src/server/cards/adventures/StoryTeller.js

```diff
@@ -0,0 +1,22 @@
+import Card from 'cards/Card';
+
+export default class StoryTeller extends Card {
+  static cost = new Card.Cost({ coin: 5 });
+  static types = new Set('Action');
+  async onPlay(player) {
+    player.actions++;
+    player.money++;
+    const cards = await player.selectCards({
+      min: 0,
+      max: 3,
+      predicate: c => c.types.has('Treasure'),
+      message: 'Choose up to three treasures to play',
+    });
+    for (const card of cards) {
```

Comment:
Player needs to be able to choose what order they play the treasures, similar to the normal buy phase. Order matters.

As an aside, what happens if someone plays Crown? It's a treasure, but it's the action phase so it works as a throne room, could someone then throne something out mid story teller?
Imagine your hand was Crown, story teller, militia. You would want to militia before story teller for the draw but you don't have actions for it, so instead you cheat it out with a crown that costs no actions because it was played with story teller.